utils/wasm-builder/src/stack_end.rs

```diff
@@ -99,11 +81,323 @@ pub fn insert_stack_end_export(
+        .global_section()
+        .ok_or("Cannot find globals section")?
+        .entries()
+        .iter()
+        .enumerate()
+    {
+        if !global.global_type().is_mutable() {
+            continue;
+        }
+        if index == data_end_index as usize {
+            continue;
+        }
+        if index == stack_pointer_index as usize {
+            continue;
+        }
+
+        let global_initial_data = handle_global_init_data(
+            global,
+            |c| c.to_le_bytes().to_vec(),
+            |c| c.to_le_bytes().to_vec(),
+        )?;
+        mut_globals.push((index, global_initial_data));
+    }
+
+    log::trace!("mutable globals are {:?}", mut_globals);
+
+    let data_end_offset = handle_global_init_data(
+        module
+            .global_section()
+            .expect("Cannot find globals section")
+            .entries()
+            .get(data_end_index as usize)
+            .expect("We have already find data end global earlier"),
+        |c| Ok(c as u32),
+        |_| Err("Wrong data section initial data instruction"),
+    )??;
+
+    log::trace!("data section end offset == {:#x}", data_end_offset);
+
+    let mut own_module = module.clone();
+    let mut global_data_offset = data_end_offset;
+    let mut new_data_in_section = vec![];
+    for (index, data) in mut_globals {
+        // Make function to get global
+        own_module = append_get_global_function(own_module, global_data_offset, data.len());
+        let get_global_function_index = (own_module
+            .functions_space()
+            .checked_sub(1)
+            .expect("Must be already at least one function"))
+            as u32;
+
+        // Make function to set global
+        own_module = append_set_global_function(own_module, global_data_offset, data.len());
+        let set_global_function_index = (own_module
+            .functions_space()
+            .checked_sub(1)
+            .expect("Must be at least one function already"))
+            as u32;
+
+        log::trace!(
+            "make get/set global functions {} and {} for global {}",
+            get_global_function_index,
+            set_global_function_index,
+            index
+        );
+
+        // Bypass all instructions in module and replace global.get and global.set
+        // by corresponding functions call.
+        for instr in own_module
+            .code_section_mut()
+            .ok_or("Cannot find code section")?
+            .bodies_mut()
+            .iter_mut()
+            .flat_map(|body| body.code_mut().elements_mut().iter_mut())
+        {
+            let global_index = u32::try_from(index).expect("Global index bigger than u32");
+            if *instr == Instruction::GetGlobal(global_index) {
+                *instr = Instruction::Call(get_global_function_index);
+            } else if *instr == Instruction::SetGlobal(global_index) {
+                *instr = Instruction::Call(set_global_function_index);
+            }
```

Comment:
Why not expressions?

```suggestion
            *instr = if *instr == Instruction::GetGlobal(global_index) {
                Instruction::Call(get_global_function_index)
            } else if *instr == Instruction::SetGlobal(global_index) {
                Instruction::Call(set_global_function_index)
            }
```

Reply:
Because instr must be changed not in all cases